### PR TITLE
enhancement(sinks): Make TLS options configurable for AWS sinks

### PR DESCRIPTION
**Pull request by ChinYing-Li**
The PR aims to address #4516 ; I will update the docs and `.cue` files once everything looks good to people.
Any suggestion is appreciated!
Signed-off-by: ChinYing-Li <chinying.li@mail.utoronto.ca>

### CODE REVIEW

**Comment by fuchsnj** [comment]
The code changes look fine. Please update the documentation for the sinks this impacted to add the new `TLS` options. You can find an example here: https://github.com/vectordotdev/vector/blob/master/website/cue/reference/components/sinks/splunk_hec_logs.cue#L44-L49

Once you push the changes, the `netlify` bot will give you a preview of the doc changes.

File: src/sinks/aws_cloudwatch_logs/mod.rs [hunk 1]
```diff
@@ -193,7 +197,7 @@ impl SinkConfig for CloudwatchLogsSinkConfig {
         let log_group = self.group_name.clone();
         let log_stream = self.stream_name.clone();
 
-        let client = self.create_client(cx.proxy())?;
+        let client = self.create_client( cx.proxy())?;
```
Comment (bruceg) [inline src/sinks/aws_cloudwatch_logs/mod.rs:200]: This added space seems accidental.

File: src/sinks/aws_cloudwatch_logs/mod.rs [hunk 2]
```diff
@@ -800,7 +804,7 @@ mod tests {
             stream: "stream".into(),
             group: "group".into(),
         };
-        let client = config.create_client(&ProxyConfig::from_env()).unwrap();
+        let client = config.create_client( &ProxyConfig::from_env()).unwrap();
```
Comment (bruceg) [inline src/sinks/aws_cloudwatch_logs/mod.rs:807]: Ditto

**Review by bruceg — CHANGES_REQUESTED** [review]
I agree with @fuchsnj, the code looks fine but this needs docs. I noted a couple of issues that the format check is complaining about, so you'll need to run `cargo fmt` to fix them.

**Review by fuchsnj — APPROVED** [review]

**Review by bruceg — APPROVED** [review]

**Comment by jszwedko** [comment]
Thank you for this contribution @ChinYing-Li !

**Pull request merged**